src/query-builder.ts

```diff
@@ -1,11 +1,12 @@
-var _ = require('lodash');
+// TODO: finish adding types to this
```

Comment:
This file is the only one I kinda gave up on for the moment. Mostly because the algorithm in `createQuery` is pretty weird.

I think we can clean it up by using a `reduce` instead of `map`ping into functions that mutate the main query object.